src/Command/DrupalUpdate/UpdateScript.php

```diff
@@ -0,0 +1,183 @@
+
+class UpdateScript
+{
+  /**
+   * Ignore files and folders list.
+   * @var string[]
+   */
+  private $ignore_updates_files = [
+        '.gitignore','.htaccess','CHANGELOG.txt','sites',
+    ];
+
+  /**
+   * Drupal directory folder path.
+   * @var string
+   */
+  private $dir;
+
+  /**
+   * @param string $dir
+   */
+  public function setDir(string $dir): void {
+    if (!defined('DRUPAL_ROOT')) {
+      define('DRUPAL_ROOT', $dir);
+    }
+    $this->updateScriptUtility->setDrupalRootDirPath();
+    $this->dir = DRUPAL_ROOT . "/docroot";
+  }
+
+  /**
+   * @var CheckInfo
+   */
+  private $checkinfo_obj;
+
+  /**
+   * @var UpdateScriptUtility
+   */
+  private $updateScriptUtility;
+
+  /**
+   * UpdateScript constructor.
+   *
+   */
+  public function __construct() {
+    $this->checkinfo_obj = new CheckInfo();
+    $this->updateScriptUtility = new UpdateScriptUtility();
+    $this->setDir(getcwd());
+  }
+
+  /**
+   * @param array $latest_security_updates
+   */
+  public function updateAvailableUpdates(array $latest_security_updates) {
+    if(count($latest_security_updates)>1){
+      // if get any available then list out in tabular format.
+      $this->updateScriptUtility->printReleaseDetail($latest_security_updates);
+
+      // update current repo with updated code based on download and folder replace method on same path.
+      $this->updateScriptUtility->updateCode($latest_security_updates);
+
+      // remove the unwanted tar or downloaded files.
+      $this->updateScriptUtility->unlinktarfiles($latest_security_updates);
+    }else{
+      // If no updates found.
+      echo "\n Branch Upto date \n";
+    }
+  }
+
+  /**
+   * Scan the directory and sub directory recorsive.
+   * Get the list of all .info files list with path.
+   * @param string $dir
+   */
+  public function getInfoFilesList() {
+    $dir = $this->dir;
+    $scaned_files = array_diff(scandir($dir), ['.', '..']);
+    foreach($scaned_files as $child_dir){
+      $temp_path = $dir . "/" . $child_dir;
+      if(is_dir($temp_path)){
+        $scanned_file_path = array_diff(scandir($temp_path), ['.', '..']);
+        $this->checkinfo_obj->getFilesInfo($scanned_file_path, $temp_path);
+      }elseif($this->checkinfo_obj->endsWith($child_dir, '.info')){
+        if(isset($this->checkinfo_obj->info_files[$child_dir])){
+          if(is_array($this->checkinfo_obj->info_files[$child_dir])){
+            $this->checkinfo_obj->info_files[$child_dir][] = $dir . "/" . $child_dir;
+          }else{
+            $this->checkinfo_obj->info_files[$child_dir][] = $this->checkinfo_obj->info_files[$child_dir];
+            $this->checkinfo_obj->info_files[$child_dir] = $dir . "/" . $child_dir;
+          }
+        }else{
+          $this->checkinfo_obj->info_files[$child_dir] = $dir . "/" . $child_dir;
+        }
+
+      }
+    }
+  }
+
+  /**
+   * Get the detail information of info files
+   */
+  public function getPackageDetailInfo() {
```

Comment:
Again, make this testable my passing in local variables. Similar to why we used dependency injection as a pattern.